projects/public-search/src/app/patron-profile/patron-profile-loans/patron-profile-loan/patron-profile-loan.component.ts

```diff
@@ -67,19 +76,29 @@ export class PatronProfileLoanComponent implements OnInit {
    * @param _translateService - TranslateService
    * @param _toastService - ToastrService
    * @param _patronProfileMenuService - PatronProfileMenuService
+   * @param _patronProfileService - PatronProfileService
    */
   constructor(
     private _loanApiService: LoanApiService,
     private _translateService: TranslateService,
     private _toastService: ToastrService,
-    private _patronProfileMenuService: PatronProfileMenuService
+    private _patronProfileMenuService: PatronProfileMenuService,
+    private _patronProfileService: PatronProfileService
   ) {}
 
   /** OnInit hook */
   ngOnInit(): void {
     this._loanApiService
       .canExtend(this.record.metadata.pid)
       .subscribe((response: CanExtend) => this.canExtend = response);
+    if (this.record.metadata.is_late) {
```

Comment:
you should have a `.is_overdue` flag that could be more relevant in this case.

`is_late` : is enddate < currentTimestamp
`is_overdue`: is enddate < currrentTimestamp && relatedCipo.firstFeeStep <= currentTimestamp

By exemple, a cipo can define a 2 days "pardon" period. 
If patron return the book at J+1 : `is_late=true` and `is_overdue=false`
If patron return the book at J+3 : `is_late=true` and `is_overdue=true`

Reply:
The flag was not called `is_overdue`, but simply `overdue` 😉. I've changed that.